### PR TITLE
Export PathStruct from AnyQuery interface

### PR DESCRIPTION
**Pull request by DanG100**
Allows users to get the PathStruct (usefully both comparing paths or printing strings)

### CODE REVIEW

File: ygnmi/types.go [hunk 1]
```diff
@@ -167,12 +167,12 @@ func (lq *leafBaseQuery[T]) isState() bool {
 }
 
 // pathStruct returns the path struct containing the path for the Query.
```
Comment (greg-dennis) [inline ygnmi/types.go:169]: pathStruct -> PathStruct

File: ygnmi/types.go [hunk 2]
```diff
@@ -253,7 +253,7 @@ func (lq *nonLeafBaseQuery[T]) isState() bool {
...
```
Comment (greg-dennis) [inline ygnmi/types.go:255]: pathStruct -> PathStruct

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: ygnmi/types.go [hunk 3]
```diff
@@ -167,12 +167,12 @@ func (lq *leafBaseQuery[T]) isState() bool {
...
-func (lq *leafBaseQuery[T]) pathStruct() PathStruct {
+func (lq *leafBaseQuery[T]) PathStruct() PathStruct {
 	return lq.ps
 }
 
 // subPaths returns the path structs used for creating the gNMI subscription.
-// A leaf query must subscribe to the schema path, so this returns the pathStruct().
+// A leaf query must subscribe to the schema path, so this returns the PathStruct().
```
[the remark this answers is not in the document]
Comment (wenovus) [reply ygnmi/types.go:175]: I think this is being said in contrast to `nonLeafBaseQuery`'s subpaths() where a batch operation could have multiple of these.

**Review by wenovus — APPROVED** [review]

File: ygnmi/types.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanG100) [reply ygnmi/types.go:175]: changed to A leaf subscribes to the same path returned by PathStruct().

**Review by greg-dennis — APPROVED** [review]

**Pull request merged**